src/libraries/Common/src/Roslyn/CSharpSyntaxHelper.cs

```diff
@@ -104,5 +104,22 @@ public override void AddAliases(CompilationOptions compilation, ref ValueListBui
             // C# doesn't have global aliases at the compilation level.
             return;
         }
+
+        public override bool ContainsGlobalAliases(SyntaxNode root)
```

Comment:
previously we would keep a table of all syntax-trees in a compilation.  With projects easily having thousands of files, this caused huge tables to be built that we would have to incrementally reprocess on each keystroke.